### PR TITLE
Hide recording tabs when recording is disabled by admin

### PR DESCRIPTION
**Pull request by Mariam05**
Closes issue #5198

### CODE REVIEW

File: app/javascript/components/rooms/room/presentation/Presentation.jsx [hunk 1]
```diff
@@ -31,14 +31,14 @@ import { PRESENTATION_MAX_FILE_COEFF, PRESENTATION_SUPPORTED_EXTENSIONS } from '
...
-  const { data: room } = useRoom(friendlyId);
+  const { data: room, isLoading } = useRoom(friendlyId);
   const { onSubmit } = useUploadPresentation(friendlyId);
 
   const onDrop = (files) => {
     onSubmit(files[0]);
   };
 
-  if (!room.presentation_name) {
+  if (isLoading || !room.presentation_name) {
```
Comment (farhatahmad) [inline app/javascript/components/rooms/room/presentation/Presentation.jsx:41]: Whats this extra check for?

File: app/controllers/api/v1/rooms_configurations_controller.rb [hunk 2]
```diff
@@ -33,6 +33,18 @@ def index
...
+      # Fetches and returns the value of the passed in configuration
+      def show
+        config_value = RoomsConfiguration.joins(:meeting_option)
+                                         .find_by(
+                                           provider: current_provider,
+                                           meeting_option: { name: params[:name] }
+                                         ).value
+
+        render_data data: config_value, status: :ok
+      end
```
Comment (farhatahmad) [inline app/controllers/api/v1/rooms_configurations_controller.rb:47]: Needs some tests :)

File: app/javascript/hooks/queries/rooms/useRoomConfigValue.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import { useQuery } from 'react-query';
+import axios from '../../../helpers/Axios';
+
+export default function useRoomConfigValue(name) {
+  return useQuery(
+    ['getRoomsConfigValue', name],
+    () => axios.get(`/rooms_configurations/${name}.json`).then((resp) => resp.data.data),
+  );
+}
```
Comment (farhatahmad) [inline app/javascript/hooks/queries/rooms/useRoomConfigValue.jsx:25]: Very nice :fire:

File: app/javascript/components/rooms/room/presentation/Presentation.jsx (hunk 1 above)
Comment (Mariam05) [reply app/javascript/components/rooms/room/presentation/Presentation.jsx:41]: When I changed the landing page for the room to `presentation`, an error was being thrown saying that `room` was undefined. Making sure it's loaded first fixed that.

Comment (farhatahmad) [reply app/javascript/components/rooms/room/presentation/Presentation.jsx:41]: You can just do `!room?.presentation_name`

File: app/controllers/api/v1/rooms_configurations_controller.rb [hunk 4]
```diff
@@ -19,7 +19,7 @@
 module Api
   module V1
     class RoomsConfigurationsController < ApiController
-      before_action only: %i[index] do
+      before_action do
```
Comment (scouillard) [inline app/controllers/api/v1/rooms_configurations_controller.rb:22]: No big deal - we settled to explicitly allow each action for more security. 
So you could add show instead.
`before_action only: %i[index show]`

File: app/javascript/components/rooms/room/presentation/Presentation.jsx (hunk 1 above)
Comment (scouillard) [reply app/javascript/components/rooms/room/presentation/Presentation.jsx:41]: Adding the safe operator `?` after every attribute solves many errors

File: spec/controllers/rooms_configurations_controller_spec.rb [hunk 5]
```diff
@@ -46,4 +46,24 @@
...
+      create(:rooms_configuration, meeting_option: create(:meeting_option, name: 'record'), value: 'false')
+    end
+
+    it 'returns the correct configuration value' do
+      get :show, params: { name: 'record' }
+
+      expect(JSON.parse(response.body)['data']).to eq('false')
+
+      expect(response).to have_http_status(:ok)
+    end
```
[the remark this answers is not in the document]
Comment (farhatahmad) [reply spec/controllers/rooms_configurations_controller_spec.rb:61]: Just 1 test returning the value is good enough i think

**Review by farhatahmad — APPROVED** [review]

**Pull request merged**